app/org/sagebionetworks/bridge/services/ConsentServiceImpl.java

```diff
@@ -168,19 +175,74 @@ public boolean hasUserSignedMostRecentConsent(StudyIdentifier studyIdentifier, U
     }
 
     @Override
-    public void withdrawConsent(Study study, User user) {
+    public void withdrawConsent(Study study, User user, Withdrawal withdrawal) {
         checkNotNull(study, Validate.CANNOT_BE_NULL, "study");
         checkNotNull(user, Validate.CANNOT_BE_NULL, "user");
-        if (userConsentDao.withdrawConsent(user.getHealthCode(), study)) {
-            decrementStudyEnrollment(study);
-            Account account = accountDao.getAccount(study, user.getEmail());
-            account.setConsentSignature(null);
+        checkNotNull(user, Validate.CANNOT_BE_NULL, "withdrawal");
+        
+        String externalId = optionsService.getOption(user.getHealthCode(), ParticipantOption.EXTERNAL_IDENTIFIER);
```

Comment:
Minor: If the call to withdraw fails (or if the user is already not consented), we can save a DDB call by moving this to after the withdraw consent call (since we don't need it until we construct the email).